content/usgs-packages/dataRetrieval_Discovery.Rmd

```diff
@@ -0,0 +1,296 @@
+
+```{r sites_sc_stream_temp, message=FALSE, warning=FALSE}
+sites_sc_stream_temp <- whatNWISsites(stateCd="SC", siteType="ST",
+                                      parameterCd="00010")
+nrow(sites_sc_stream_temp)
+```
+
+We are now down to just `r prettyNum(nrow(sites_sc_stream_temp), big.mark=",")` sites, much less than our original `r prettyNum(nrow(sites_sc), big.mark=",")`. To actually download this data, you can query using our three arguments, `stateCd` + `siteType` + `parameterCd`, or by the site numbers from the `sites_sc_stream_temp` data.frame using `unique(sites_sc_stream_temp[['site_no']])`. Downloading NWIS data will be covered in the next section, [readNWISdata](/usgs-packages/dataRetrieval-readNWIS).
+
+The `whatNWISsites` function can also be very useful for making quick maps with site locations, see the columns `dec_lat_va` and `dec_long_va` (decimal latitude and longitude value). For instance,
+
+```{r sc_streamtemp_sites_map, message=FALSE, warning=FALSE, fig.cap="Geographic locations of NWIS South Carolina stream sites with temperature data"}
+# SC stream temperature sites
+library(maps)
+map('state', regions='south carolina')
+title(main="South Carolina Stream Temp Sites")
+points(x=sites_sc_stream_temp$dec_long_va, 
+       y=sites_sc_stream_temp$dec_lat_va)
+```
+
+You can also quickly compare the amount of data between states:
+
+```{r sites_scVSwv_stream_temp, message=FALSE, warning=FALSE}
+# stream temperature data in SC vs WV
+sites_wv_stream_temp <- whatNWISsites(stateCd="WV", siteType="ST",
+                                      parameterCd="00010")
+nrow(sites_sc_stream_temp)
+nrow(sites_wv_stream_temp)
+```
+
+Or compare the amount of data between site types:
+
+```{r sites_sc_streamVSlake_temp, message=FALSE, warning=FALSE}
+# SC temperature data in streams vs lakes
+sites_sc_lake_temp <- whatNWISsites(stateCd="SC", siteType="LK",
+                                      parameterCd="00010")
+nrow(sites_sc_stream_temp)
+nrow(sites_sc_lake_temp)
+```
+
+What if you wanted sites from multiple states? You need to separately query each state because `stateCd` can only be length 1. You can use for loops or the `apply` family of functions to accomplish this. The example here uses `lapply`, `Reduce`, and the `dplyr` function, `full_join`. We will assume you know how to use these functions. 
+
+```{r sites_mult_states_stream_temp, message=FALSE, warning=FALSE}
+# South West US sites with stream temperature
+sites_sw_stream_temp <- lapply(c("CA", "AZ", "NM", "NV"), function(st){
+  whatNWISsites(stateCd=st, siteType="ST", parameterCd="00010")
+})
+
+# combine the returned list into one using dplyr::full_join
+library(dplyr)
+sites_sw_stream_temp <- Reduce(full_join, sites_sw_stream_temp)
```

Comment:
I agree - I think I did lapply because people always talk about not using for loops. I did have to do some googling to get to that solution, so it was not intuitive :)